include/dr/views/transform.hpp

```diff
@@ -23,12 +20,12 @@ class transform_view : public rng::view_interface<transform_view<V, F>> {
   auto end() const {
     auto segs = segments();
     return normal_distributed_iterator<decltype(segments())>(
-        segs, size_t(segs.size()), 0);
+        segs, std::size_t(segs.size()), 0);
```

Comment:
@BenBrock: Some views do not have a size. Also, shouldn't the last argument be the size of the last segment?

Reply:
Can we guarantee that the segments view itself will have a size, even if all the individual segments themselves don't?

(i.e. can we say the thing you get back from `segments()` is a `sized_range`?)

If not, I think we can at least say it's a `forward_range`, and then we can check `std::distance(first, last)`, although I think we can probably say it's a sized range?

As for the end iterator, the way it's implemented, the `segment_index` will be incremented as soon as you reach the end of a segment, including the last segment, so the iterator you get at one past the final element is actually the 0'th element of one past the last segment.

Reply:
I don't remember whether it was the segments or the segment which did not have a size. I don't understand why my mhp segments don't have a size. It looks like there is zip view with an unbounded iota inside. I was already planning to reimplement distributed_vector so it directly provides segments and relies on normal_distributed_iterator to provide the flat view, so I will do that first. Also, we should reimplement the other view code to be like transform_view. After that, the problem might go away. Or at least we would have more common code between shp and mhp.